src/test/java/org/folio/edge/rtac/utils/RtacOkapiClientCompressionTest.java

```diff
@@ -0,0 +1,77 @@
+package org.folio.edge.rtac.utils;
+
+import java.util.UUID;
+
+import org.apache.log4j.Logger;
+import org.folio.edge.core.utils.test.TestUtils;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import io.vertx.core.MultiMap;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpServer;
+import io.vertx.core.http.HttpServerOptions;
+import io.vertx.ext.unit.Async;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+
+@RunWith(VertxUnitRunner.class)
+public class RtacOkapiClientCompressionTest {
+  private static final Logger logger = Logger.getLogger(RtacOkapiClientCompressionTest.class);
+
+  private final Vertx vertx = Vertx.vertx();
+
+  private final String titleId = UUID.randomUUID().toString();
+  private static final String tenant = "diku";
+  private static final long reqTimeout = 3000L;
+
+  private RtacOkapiClient client;
+
+  @Before
+  public void setUp(TestContext context) throws Exception {
+    int okapiPort = TestUtils.getPort();
+
+    final HttpServer server = vertx.createHttpServer(
+        new HttpServerOptions().setCompressionSupported(true));
+
+    server.requestHandler(req -> {
+      req.response()
+        .setStatusCode(200)
+        .putHeader("content-type", "application/json")
+        .end("{\"test\":\"1234\"}");
+    });
+
+    final Async async = context.async();
+    server.listen(okapiPort, "localhost", ar -> {
+      context.assertTrue(ar.succeeded());
+      async.complete();
+    });
+
+    client = new RtacOkapiClientFactory(vertx,
+        "http://localhost:" + okapiPort, reqTimeout).getRtacOkapiClient(tenant);
+  }
+
+  @After
+  public void tearDown(TestContext context) {
+    vertx.close(context.asyncAssertSuccess());
+  }
+
+  @Test
```

Comment:
Can you please explain how this test achieves the goal stated in the JIRA: "add a test that breaks with the old version (no compression) and works with 2.0.2."?  The test sets up the server manually, not using  the code that I assume to exist in edge-common v2.0.2.. . Anything that works here doesn't prove that it works with 2.0.2.  Am I reasoning incorrectly?

Reply:
The server is not part of edge-common, only the client. So, what we are doing is creating a server that uses compression, like Okapi. Then we create a edge-common based RTAC client that wants the server to send compressed data, if it can. If we use the 2.0.1 version of edge-common, this test fails due to compressed data being in the response body. If we change the version to edge-common 2.0.2, then this test passes. If someone messes with client compression in edge-common and releases a new version, this test should at least prove or disprove that compression will work when we upgrade to the new version.

Does that make sense?

Reply:
So getting the clear text data in the body means that the message was decompressed by the 2.0.2 client?

Reply:
Exactly.

Reply:
Can you please add a description/comment somewhere in this test to describe this behavior? I'm afraid someone who isn't familiar with the code wouldn't know what it's doing.

Reply:
I added a comment on the test. Do you think it helps?

Reply:
Yes, it's helpful. Thanks.